### PR TITLE
Feature/logging

### LINKED ISSUE #57: Better logging
Maybe something like:
https://github.com/winstonjs/winston

Used universally.

### PR DESCRIPTION
**Pull request by bayleedev**
This adds universal logging to zazu Fixes #57
### Levels
- Logged for verbose changes is non-interaction
- Logging for info is interaction changes
- Logging for warn is dangerous
- Logging for error is an error
### Features
- Plugin Debugger works per usual, with the new log levels
- Documentation is updated in the PR to reflect the logging api change
- Cucumber specs output log on failures
### Bug Fixes

The `eventually` method was doing some crazy stuff with promises, I refactored it so it didn't output as much junk upon failures.

### CODE REVIEW

File: app/background.js [hunk 1]
```diff
@@ -4,13 +4,15 @@ const path = require('path')
...
 const globalEmitter = require('./lib/globalEmitter')
+const logger = require('./lib/logger')
 
 const { windowHelper, openCount } = require('./helpers/window')
 const forceSingleInstance = require('./helpers/singleInstance')
 const addToStartup = require('./helpers/startup')
 const createMenu = require('./helpers/menu')
 
 globalEmitter.on('showDebug', (message) => {
+  logger.log('info', 'opening about page')
```
Comment (bayleedev) [inline app/background.js]: opening debug page

File: app/helpers/startup.js [hunk 2]
```diff
@@ -12,11 +13,13 @@ module.exports = () => {
 
       appLauncher.isEnabled().then((enabled) => {
         if (enabled) return
+        logger.log('info', 'Adding to linux startup')
         return appLauncher.enable()
       })
     } else {
       const settings = app.getLoginItemSettings()
       if (!settings.openAtLogin) {
+        logger.log('info', 'Adding to win32 or darwin startup')
```
Comment (bayleedev) [inline app/helpers/startup.js:22]: These should be verbose

File: app/packages/package.js [hunk 3]
```diff
@@ -4,17 +4,20 @@ const jetpack = require('fs-jetpack')
...
     this.path = path.join(configuration.pluginDir, url)
     this.url = url
     this.clone = clone
     this.pull = pull
+    this.logger = logger.bindMeta({ plugin: this.url })
   }
 
   load () {
     return this.download().then(() => {
+      this.logger.log('info', 'loading package')
```
Comment (bayleedev) [inline app/packages/package.js]: verbose

File: app/packages/package.js [hunk 4]
```diff
@@ -28,13 +31,15 @@ class Package {
...
     }
+    this.logger.log('info', 'update plugin')
     return this.pull(this.path)
   }
 
   download () {
     if (jetpack.exists(this.path)) {
       return Promise.resolve('exists')
     }
+    this.logger.log('info', 'cloning plugin')
```
Comment (bayleedev) [inline app/packages/package.js]: verbose

File: app/packages/plugin.js [hunk 5]
```diff
@@ -26,20 +26,23 @@ class Plugin extends Package {
...
     this.activeState = activeState
+    if (this.activeState) this.logger.log('info', 'scoping plugin')
     this.inputs.forEach((input) => {
       input.setScoped(input.id === blockId)
     })
   }
 
   update () {
     return super.update().then(() => {
+      this.logger.log('info', 'npm install')
```
Comment (blainesch) [inline app/packages/plugin.js]: verbose

File: app/helpers/startup.js (hunk 2 above)
Comment (bayleedev) [inline app/helpers/startup.js:22]: actually they could be info if it was triggered by a user, checking is silly so keep it as info.

**Comment by afaur** [comment]
👍

**Pull request merged**